internal/rest/types/server.go

```diff
@@ -6,9 +6,10 @@ import (
 
 // Server represents server status information.
 type Server struct {
-	Name    string         `json:"name"    yaml:"name"`
-	Address types.AddrPort `json:"address" yaml:"address"`
-	Ready   bool           `json:"ready"   yaml:"ready"`
+	Name          string         `json:"name"    yaml:"name"`
+	Address       types.AddrPort `json:"address" yaml:"address"`
+	Ready         bool           `json:"ready"   yaml:"ready"`
+	APIExtensions []string       `json:"api_extensions" yaml:"api_extensions"`
```

Comment:
Hmm, so does this mean that on the other end, checking for a particular extension means iterating through this list like any other slice?

Could we implement the JSON/YAML marshal/unmarshal commands and then pass this through as a real `Extensions` type so that the other end of the API request can use the Extension methods to check if this set contains a particular extension?

Reply:
Here's a reference for the `AddrPort` type: 
https://github.com/canonical/microcluster/blob/a1a192380cd5a3f534ba0d48bf7a838ef484d359/rest/types/addrport.go#L41-L82

Reply:
the `Extensions` type is an `internal` type so we'll have to move `Extensions` out of `internal/` if we want to do that (which I'm happy to do if you think its best)

Reply:
How come? The user won't have to actually declare the type to use methods from it.

Reply:
hmmm.. you're right.. sorry for that. I'm reverting the change I just pushed

Reply:
```go
package main

import (
	"context"
	"fmt"

	"github.com/canonical/microcluster/microcluster"
)

func main() {
	m, _ := microcluster.App(microcluster.Args{})

	server, _ := m.Status(context.Background())

        // Call .HasExtension without importing anything.
	fmt.Println(server.Extensions.HasExtension("some-ext"))
}
```

Reply:
@masnax updated :)